stratum/hal/lib/tdi/dpdk/dpdk_hal.cc

```diff
@@ -19,6 +19,7 @@
 #include "stratum/lib/constants.h"
 #include "stratum/lib/macros.h"
 #include "stratum/lib/utils.h"
+#include "krnlmon_main.h"
```

Comment:
This creates an unconditional dependency on krnlmon, meaning that we can no longer build stratum without it.

I think the krnlmon code should be conditionalized (e.g. `#ifdef WITH_KRNLMON`). You'd need to add:
```
if(WITH_KRNLMON)
    target_compile_definitions(stratum_main_o PRIVATE WITH_KRNLMON)
endif()
```
to stratum/CMakeLists.txt.

Reply:
Done

Reply:
My recommendation was that you not use this module at all if krnlmon isn't enabled. You've opted for a messier, more difficult to maintain solution, over a simple one.